utils/date.ts

```diff
@@ -0,0 +1,34 @@
+const SECOND = 1_000
+const MINUTE = SECOND * 60
+const HOUR = MINUTE * 60
+const DAY = HOUR * 24
+
+export function timeAgo(timestamp: number) {
```

Comment:
I've noticed that we're dealing with a lot of date formatting and its calculations, and it might be better to consider [date-fns](https://date-fns.org/) if it's likely to continue this trend.

Reply:
I did intentionally create a custom function for formatting datetimes to avoid using date-fns as it is large, 6.61 MB total! But I have to deal with formatting date times quite many times, I will refactor it with date-fns. Thanks for the comment!